bindata/network/ovn-kubernetes/ovnkube-cno-master.sh

```diff
@@ -0,0 +1,770 @@
+	    export OVN_NB_DAEMON=$(ovn-nbctl --pidfile=/var/run/ovn/ovn-nbctl.pid \
+		    --detach \
+		    -p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt \
+		    --db "${OVN_NB_DB_LIST}")
+
+	    # REMOVEME once OVN path for control socket is fixed (right now uses /var/run/openvswitch)
+	    ln -sf $OVN_NB_DAEMON /var/run/ovn/ || true
+
+	    echo "I$(date "+%m%d %H:%M:%S.%N") - ovnkube-master - start ovnkube --init-master ${K8S_NODE}"
+	    exec /usr/bin/ovnkube \
+		    --init-master "${K8S_NODE}" \
+		    --config-file=/run/ovnkube-config/ovnkube.conf \
+		    --ovn-empty-lb-events \
+		    --loglevel "${OVN_KUBE_LOG_LEVEL}" \
+		    --metrics-bind-address "127.0.0.1:29102" \
+		    ${gateway_mode_flags} \
+		    --sb-address "${OVN_SB_DB_LIST}" \
+		    --sb-client-privkey /ovn-cert/tls.key \
+		    --sb-client-cert /ovn-cert/tls.crt \
+		    --sb-client-cacert /ovn-ca/ca-bundle.crt \
+		    --sb-cert-common-name "${OVN_CERT_CN}" \
+		    --nb-address "${OVN_NB_DB_LIST}" \
+		    --nb-client-privkey /ovn-cert/tls.key \
+		    --nb-client-cert /ovn-cert/tls.crt \
+		    --nb-client-cacert /ovn-ca/ca-bundle.crt \
+		    --nbctl-daemon-mode \
+		    --nb-cert-common-name "${OVN_CERT_CN}" \
+		    --enable-multicast
+    }   
+
+    run-ovnkube-master-preStop() {
+	    kill $(cat /var/run/ovn/ovn-nbctl.pid) && unset OVN_NB_DAEMON
+
+    }
+
+  
+   ############ Main ################ 
+          set -xe
+          if [[ -f "/env/_master" ]]; then
+            set -o allexport
+            source "/env/_master"
+            set +o allexport
+          fi
+
+ 
+    echo "================== ovnkube-cno-master.sh================"
+    
+    echo " ==================== command: ${cmd}"
+    
+    case  ${cmd} in
+	    "run-northd")
+		    run-northd
+		    ;;
+	    "run-nbdb")
+		    run-nbdb
+		    ;;
+	    "run-ovn-dbchecker")
+		    run-ovn-dbchecker
+		    ;;
+	    "run-nbdb-postStart")
+		    run-nbdb-postStart
+		    ;;
+	    "run-ovnkube-master-preStop") 
+		    run-ovnkube-master-preStop
+		    ;;
+	    "run-ovnkube-master")
+		    run-ovnkube-master
+		    ;;
+	    "run-sbdb-postStart")
+		    run-sbdb-postStart
+		    ;;
+	    "run-sbdb")
+		    run-sbdb
+		    ;;
+	    "run-sbdb-preStop") 
+		    run-sbdb-preStop
+		    ;;
+	    "run-nbdb-preStop")
+		    run-nbdb-preStop
+		    ;;
+	    "run-sbdb-readinessProbe")
+		    run-sbdb-readinessProbe
+		    ;;
+	    "run-nbdb-readinessProbe")
+		    run-nbdb-readinessProbe
+		    ;;
+	    "run-kube-rbac-proxy")
+		    run-kube-rbac-proxy
+		    ;;
+	    *)
+	    echo "invalid command ${cmd}"
+	    echo "valid commands: run-ovn-northd run-ovn-dbchecker " \
+		    "run-nbdb run-nbdb-postStart run-kube-rbac-proxy run-nbdb-readinessProbe " \ 
+		    "run-sbdb-readinessProbe run-nbdb-preStop run-sbdb-preStop run-sbdb-postStart run-sbdb run-sbdb-postStart"
+	    exit 0
+	    ;;
+    esac
+
+  exit 0 
```

Comment:
This seems to be growing rather large for a bash script.  Would it make more sense to start turning these into Go binaries?  Or is Python an alternative?

Reply:
Or maybe we can add some bash unittests for the functions. I guess there is BATS, https://github.com/sstephenson/bats

Reply:
Please lint with shellcheck

https://www.shellcheck.net/

```
ovnkube-cno-master.sh:5:1: warning: ovn_log_level_northd appears unused. Verify use (or export if used externally). [SC2034]
ovnkube-cno-master.sh:15:20: note: Not following: /env/_master was not specified as input (see shellcheck -x). [SC1091]
ovnkube-cno-master.sh:36:20: note: Not following: /env/_master was not specified as input (see shellcheck -x). [SC1091]
ovnkube-cno-master.sh:60:72: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:61:11: note: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?. [SC2181]
ovnkube-cno-master.sh:66:26: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:67:18: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:67:57: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:67:68: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:67:76: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:84:114: note: egrep is non-standard and deprecated. Use grep -E instead. [SC2196]
ovnkube-cno-master.sh:87:28: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:87:36: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:87:40: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:93:26: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:112:24: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:112:30: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:125:34: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:126:35: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:127:34: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:127:47: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:128:35: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:141:35: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:142:35: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:142:48: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:153:35: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:154:36: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:155:35: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:155:48: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:156:36: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:169:33: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:170:33: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:170:46: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:193:79: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:193:93: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:220:23: warning: The surrounding quotes actually unquote this. Remove or escape them. [SC2027]
ovnkube-cno-master.sh:227:25: note: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?. [SC2181]
ovnkube-cno-master.sh:228:51: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:239:81: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:240:27: note: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?. [SC2181]
ovnkube-cno-master.sh:273:27: note: Use -n instead of ! -z. [SC2236]
ovnkube-cno-master.sh:286:25: note: Use -n instead of ! -z. [SC2236]
ovnkube-cno-master.sh:287:59: warning: Quote the right-hand side of != in [[ ]] to prevent glob matching. [SC2053]
ovnkube-cno-master.sh:288:45: note: $/${} is unnecessary on arithmetic variables. [SC2004]
ovnkube-cno-master.sh:290:115: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:314:126: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:343:18: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:347:18: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:352:23: note: $/${} is unnecessary on arithmetic variables. [SC2004]
ovnkube-cno-master.sh:358:22: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:360:22: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:384:20: note: Not following: /env/_master was not specified as input (see shellcheck -x). [SC1091]
ovnkube-cno-master.sh:407:80: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:408:19: note: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?. [SC2181]
ovnkube-cno-master.sh:413:34: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:414:26: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:414:65: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:414:76: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:414:84: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:431:122: note: egrep is non-standard and deprecated. Use grep -E instead. [SC2196]
ovnkube-cno-master.sh:434:37: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:434:45: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:434:49: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:440:34: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:459:33: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:459:39: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:469:15: warning: initial_raft_create appears unused. Verify use (or export if used externally). [SC2034]
ovnkube-cno-master.sh:472:42: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:473:43: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:474:42: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:474:55: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:475:43: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:488:44: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:489:44: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:489:57: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:499:44: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:500:45: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:501:44: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:501:57: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:502:45: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:515:40: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:516:40: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:516:53: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:536:79: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:536:93: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:583:27: note: Use -n instead of ! -z. [SC2236]
ovnkube-cno-master.sh:596:25: note: Use -n instead of ! -z. [SC2236]
ovnkube-cno-master.sh:597:59: warning: Quote the right-hand side of != in [[ ]] to prevent glob matching. [SC2053]
ovnkube-cno-master.sh:598:45: note: $/${} is unnecessary on arithmetic variables. [SC2004]
ovnkube-cno-master.sh:600:115: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:625:117: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:632:20: note: Not following: /env/_master was not specified as input (see shellcheck -x). [SC1091]
ovnkube-cno-master.sh:658:20: note: Not following: /env/_master was not specified as input (see shellcheck -x). [SC1091]
ovnkube-cno-master.sh:672:13: warning: Declare and assign separately to avoid masking return values. [SC2155]
ovnkube-cno-master.sh:678:13: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:687:7: note: Double quote to prevent globbing and word splitting. [SC2086]
ovnkube-cno-master.sh:703:11: warning: Quote this to prevent word splitting. [SC2046]
ovnkube-cno-master.sh:712:20: note: Not following: /env/_master was not specified as input (see shellcheck -x). [SC1091]
ovnkube-cno-master.sh:764:82: error: Delete trailing spaces after \ to break line (or use quotes for literal space). [SC1101]
```

Reply:
@rbbratta Thanks for the useful comments. Fyi.. will be moving the bash script to OVN-Kubernetes repo
https://github.com/openshift/ovn-kubernetes/pull/350/